### PR TITLE
Edit crossing details

### PR DESCRIPTION
**Pull request by briaguya**
When working on the edit crossing functionality for the crossing detail page I found that we didn't have an edit crossing function on the backend. I updated the schema to add the function, then added some tests to the backend to ensure the graphql mutation works correctly. I then hooked the crossing detail component up to the mutation.

### CODE REVIEW

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js [hunk 1]
```diff
@@ -1,6 +1,34 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: props.crossing.name,
+      description: props.crossing.description
+    };
+  }
+
+  updateCrossing(e) {
+    debugger;
```
Comment (mateoclarke) [inline frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: giphy 14

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+    this.props.updateCrossingMutation({
+      variables: {
+        crossingId: this.props.crossing.id,
+        name: this.state.name,
+        description: this.state.description
+      }
+    })
+    .then(({ data }) => {
+      alert(data);
```
Comment (mateoclarke) [inline frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: This returns `[Object object]` for me. Seems like a placeholder for a success msg, but could it be more helpful in the meantime?

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js [hunk 3]
```diff
@@ -9,11 +37,11 @@ class CrossingDetails extends Component {
...
         <div> {crossing.humanCoordinates} </div>
-        <input type='text' defaultValue={crossing.name}></input>
+        <input type="text" value={this.state.name} onChange={this.nameChanged}/>
         <div> {crossing.humanAddress} </div>
-        <input type='text' defaultValue={crossing.description}></input>
-        <input type='text' defaultValue={community.name}></input>
-        <div> SAVE CROSSING </div>
+        <input type="text" value={this.state.description} onChange={this.descriptionChanged}/>
+        <div>{community.name}</div>
+        <button onClick={this.updateCrossing.bind(this)}> SAVE CROSSING </button>
```
Comment (ifsimicoded) [inline frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: Probably want to bind this to the handler as you did in CrossingListHeader.js here: https://github.com/cityofaustin/ctxfloods/blob/master/frontend/src/Dashboard/CrossingListPage/CrossingListHeader/CrossingListHeader.js#L24

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js [hunk 4]
```diff
@@ -1,6 +1,34 @@
...
+      }
+    })
+    .then(({ data }) => {
+      alert(data);
+    }).catch((error) => {
+      console.log('there was an error sending the query', error);
+    });
+  }
+
+  nameChanged = (e) => { this.setState({ name: e.target.value }) };
```
Comment (ifsimicoded) [inline frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: Not entirely sure how React works, but do these onchange handlers need to be throttled at all? Not sure if React does this out of the box, or if a new update and render cycle based on a state change is called every time a type-y type-y happens.

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js [hunk 5]
```diff
@@ -1,6 +1,34 @@
...
+    super(props);
+    this.state = {
+      name: props.crossing.name,
+      description: props.crossing.description
+    };
+  }
+
+  updateCrossing(e) {
+    debugger;
+    this.props.updateCrossingMutation({
```
Comment (ifsimicoded) [inline frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: This didn't persist changes for me on the full crossing list page. Maybe that's known, but this graphql fanciness is foreign to me, so I don't know any better than to ask

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js (hunk 1 above)
Comment (briaguya) [reply frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: whoops

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js (hunk 2 above)
Comment (briaguya) [reply frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: replaced with a todo

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js (hunk 5 above)
Comment (briaguya) [reply frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: In my testing the changes do persist, but require a refresh. This is a situation where we could call refetchqueries to get all of the crossings again, but I figured it'd be better to wait until we figured out a good solution for only refetching what we need.

I found a pattern that makes sense and implemented it on the crossing list page for status updates, moving forward it'll make sense to follow that pattern to propagate the crossing edits correctly.

File: frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js (hunk 4 above)
Comment (briaguya) [reply frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: I haven't heard about this pattern being an issue in my research. @mateoclarke - any insights?

Comment (mateoclarke) [reply frontend/src/Dashboard/CrossingDetailPage/CrossingDetails.js]: Yeah, I think you'd need to manually throttle. I googled for a sec a didn't find a clear pattern either. Good thing to be aware but maybe the performance cost isn't that significant?

**Pull request merged**